task/thermal/src/main.rs

```diff
@@ -21,14 +21,32 @@ use crate::{
 };
 use core::convert::TryFrom;
 use drv_i2c_api::ResponseCode;
-pub use drv_i2c_devices::max31790::Fan;
 use drv_i2c_devices::max31790::I2cWatchdog;
 use idol_runtime::{NotificationHandler, RequestError};
 use ringbuf::*;
 use task_thermal_api::{ThermalError, ThermalMode};
 use userlib::units::PWMDuty;
 use userlib::*;
 
+//
+// We define our own Fan type, as we may have more fans than any single
+// controller supports.
+//
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct Fan(u8);
+
+impl From<Fan> for drv_i2c_devices::max31790::Fan {
```

Comment:
Nit: I'd rather have this be `impl TryFrom...` and let the caller unwrap it, since this implementation hides the fact that it could fail.

Reply:
Agreed -- thank you!